integration_test/third_party_apps_test.go

```diff
@@ -363,6 +364,75 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "upstream/master")
+	out, err := cmd.Output()
+	if err != nil {
+		t.Fatalf("got error calling `git diff`: %v", err)
+	}
+
+	return strings.Split(string(out), "\n")
+}
+
+func determineImpactedApps(mf []string) map[string]bool {
+	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		if strings.HasPrefix(f, "apps/") {
+			// File names: apps/<appname>.go
+			f := strings.TrimPrefix(f, "apps/")
+			f = strings.TrimSuffix(f, ".go")
+			impactedApps[f] = true
+
+		} else if strings.HasPrefix(f, "integration_test/third_party_apps_data/applications/") {
+			// Folder names: integration_test/third_party_apps_data/applications/<app_name>
+			f := strings.TrimPrefix(f, "integration_test/third_party_apps_data/applications/")
+			f = strings.Split(f, "/")[0]
+			impactedApps[f] = true
+
+		} else {
+			// Any other modified file means we should test
+			// everything.
+			impactedApps["all"] = true
+			break
+		}
+
+	}
+	return impactedApps
+}
+
+type test struct {
+	platform   string
+	app        string
+	skipReason string
+}
+
+const defaultPlatform = "debian-10"
+
+func determineTestsToSkip(tests []test, impactedApps map[string]bool, testConfig testConfig) {
```

Comment:
I feel like this needs a comment. Perhaps a link to the doc (which is already shared with observiq folks) and the doc made an effort to be publicly viewable.

Reply:
Comment forthcoming.

Reply:
Done.